pkg/sink/sink.go

```diff
@@ -85,16 +85,20 @@ func createResources(resources []json.RawMessage, restClient restclient.Interfac
 
 // createResource uses the kubeClient to create the resource defined in the
 // TriggerResourceTemplate and returns any errors with this process
-func createResource(rt json.RawMessage, restClient restclient.Interface, discoveryClient discoveryclient.DiscoveryInterface) error {
+func createResource(rt json.RawMessage, restClient restclient.Interface, discoveryClient discoveryclient.DiscoveryInterface, eventListenerNamespace string) error {
 	// Assume the TriggerResourceTemplate is valid (it has an apiVersion and Kind)
 	apiVersion := gjson.GetBytes(rt, "apiVersion").String()
 	kind := gjson.GetBytes(rt, "kind").String()
 	namespace := gjson.GetBytes(rt, "metadata.namespace").String()
-	namePlural, err := findAPIResourceNamePlural(discoveryClient, apiVersion, kind)
+	// Default the resource creation to the EventListenerNamespace if not found in the resource template
+	if namespace == "" {
```

Comment:
Namespace is an [omit empty value](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L147) where it should be the same. Also, this namespace value is just passed to `createRequestURI()`, where for cluster resources it will be ignored.

Reply:
Spoke with @ncskier more on this. Unless we check for this specifically in the `TriggerTemplate` validation, both approaches have pros and cons. Currently, if the resource template explicitly declares an empty namespace, this will also be overridden to the `EventListenerNamespaces`, otherwise we would be create a malformed URI.